effekt/shared/src/main/scala/effekt/typer/CapabilityScope.scala

```diff
@@ -20,11 +22,60 @@ case object GlobalCapabilityScope extends CapabilityScope {
 }
+
+object DummyUnifier extends TypeUnifier {
+  // XXX(HACK): Side-channeling the output of `unifyValueTypes` via this variable
+  var success = true
+
+  def checkSubtype(t1: effekt.symbols.ValueType, t2: effekt.symbols.ValueType)(using C: Context): Boolean = {
+    success = true
+    unifyValueTypes(
+      C.substitution.substitute(t1),
+      C.substitution.substitute(t2),
+      ErrorContext.MergeTypes(t1, t2))
+    success
+  }
+
+  override def requireLowerBound(x: TypeVar.UnificationVar, tpe: ValueType, ctx: ErrorContext): Unit = ()
+  override def requireUpperBound(x: TypeVar.UnificationVar, tpe: ValueType, ctx: ErrorContext): Unit = ()
+  override def requireEqual(x: TypeVar.UnificationVar, tpe: ValueType, ctx: ErrorContext): Unit = ()
+  override def requireSubregion(lower: Captures, upper: Captures, ctx: ErrorContext): Unit = ()
+
+  override def abort(msg: String, ctx: ErrorContext): Nothing = sys error "Unexpected abort in DummyUnifier!" // NOTE(jiribenes, 2025-11-11): shouldn't ever be triggered!
+
+  override def error(msg: String, ctx: ErrorContext): Unit = { success = false }
+  override def error(left: Type, right: Type, ctx: ErrorContext): Unit = { success = false }
+}
+
 class BindSome(binder: source.Tree, capabilities: Map[symbols.InterfaceType, symbols.BlockParam],val parent: CapabilityScope) extends CapabilityScope {
   def copy: CapabilityScope = BindSome(binder, capabilities, parent.copy)
   def capabilityFor(tpe: symbols.InterfaceType)(using C: Context): symbols.BlockParam =
-    capabilities.getOrElse(tpe, parent.capabilityFor(tpe))
+    capabilities.get(tpe).orElse {
+      // If the requested type has unbound type parameters, try lexical resolution (see PR #1194)
+      unknowns(tpe).toList match {
+        case Nil => None
+        case _ => {
+          def isValidFit(handlerTpe: symbols.InterfaceType): Boolean =
+            handlerTpe.typeConstructor == tpe.typeConstructor &&
+              (tpe.args zip handlerTpe.args).forall { case (inferred, concrete) => DummyUnifier.checkSubtype(inferred, concrete) }
+
+          capabilities.collect { case (handlerTpe, cap) if isValidFit(handlerTpe) => (handlerTpe, cap) }.toList match {
+            case List((handlerTpe, cap)) =>
+              (tpe.args zip handlerTpe.args).foreach { case (inferred, concrete) =>
+                C.requireSubtype(inferred, concrete, ErrorContext.MergeTypes(inferred, concrete))
+              }
+              Some(cap)
+            case Nil => None
+            case caps => None
+          }
+        }
+      }
+    }.getOrElse(parent.capabilityFor(tpe))
+
+  def relevantInScopeFor(tpe: BlockType.InterfaceType)(using C: Context): Set[BlockType.InterfaceType] =
+    capabilities.collect { case (handlerTpe, _) if handlerTpe.typeConstructor == tpe.typeConstructor && unknowns(handlerTpe).isEmpty => handlerTpe }.toSet ++ parent.relevantInScopeFor(tpe)
```

Comment:
I don't understand why
```
&& unknowns(handlerTpe).isEmpty
```
is needed here.

Try the `examples/neg/lexical_capability_selection_ambiguous.effekt` example without these additional conditions, and you'll also see `Greet[T]` suggested, which is whack.